src/items.cpp

```diff
@@ -372,514 +532,788 @@ void Items::parseItemNode(const pugi::xml_node& itemNode, uint16_t id)
+			ItemParseAttributes_t parseType = parseAttribute->second;
+			switch (parseType) {
+				case ITEM_PARSE_TYPE: {
+					tmpStrValue = asLowerCaseString(valueAttribute.as_string());
+					auto it2 = ItemTypesMap.find(tmpStrValue);
+					if (it2 != ItemTypesMap.end()) {
+						it.type = it2->second;
+						if (it.type == ITEM_TYPE_CONTAINER) {
+							it.group = ITEM_GROUP_CONTAINER;
+						}
+					} else {
+						std::cout << "[Warning - Items::parseItemNode] Unknown type: " << valueAttribute.as_string() << std::endl;
+					}
+					break;
+				}
+
+				case ITEM_PARSE_DESCRIPTION: {
+					it.description = valueAttribute.as_string();
+					break;
+				}
+
+				case ITEM_PARSE_RUNESPELLNAME: {
+					it.runeSpellName = valueAttribute.as_string();
+					break;
+				}
+
+				case ITEM_PARSE_WEIGHT: {
+					it.weight = pugi::cast<uint32_t>(valueAttribute.value());
+					break;
+				}
+
+				case ITEM_PARSE_SHOWCOUNT: {
+					it.showCount = valueAttribute.as_bool();
+					break;
+				}
+
+				case ITEM_PARSE_ARMOR: {
+					it.armor = pugi::cast<int32_t>(valueAttribute.value());
+					break;
+				}
+
+				case ITEM_PARSE_DEFENSE: {
+					it.defense = pugi::cast<int32_t>(valueAttribute.value());
+					break;
+				}
+
+				case ITEM_PARSE_EXTRADEF: {
+					it.extraDefense = pugi::cast<int32_t>(valueAttribute.value());
+					break;
+				}
+
+				case ITEM_PARSE_ATTACK: {
+					it.attack = pugi::cast<int32_t>(valueAttribute.value());
+					break;
+				}
+
+				case ITEM_PARSE_ROTATETO: {
+					it.rotateTo = pugi::cast<int32_t>(valueAttribute.value());
+					break;
+				}
+
+				case ITEM_PARSE_MOVEABLE: {
+					it.moveable = valueAttribute.as_bool();
+					break;
+				}
+
+				case ITEM_PARSE_BLOCKPROJECTILE: {
+					it.blockProjectile = valueAttribute.as_bool();
+					break;
+				}
+
+				case ITEM_PARSE_PICKUPABLE: {
+					it.allowPickupable = valueAttribute.as_bool();
+					break;
+				}
+
+				case ITEM_PARSE_FLOORCHANGE: {
+					tmpStrValue = asLowerCaseString(valueAttribute.as_string());
+					auto it2 = TileStatesMap.find(tmpStrValue);
+					if (it2 != TileStatesMap.end()) {
+						it.floorChange = it2->second;
+					} else {
+						std::cout << "[Warning - Items::parseItemNode] Unknown floorChange: " << valueAttribute.as_string() << std::endl;
+					}
+					break;
+				}
+
+				case ITEM_PARSE_CORPSETYPE: {
+					tmpStrValue = asLowerCaseString(valueAttribute.as_string());
+					auto it2 = RaceTypesMap.find(tmpStrValue);
+					if (it2 != RaceTypesMap.end()) {
+						it.corpseType = it2->second; 
+					} else {
+						std::cout << "[Warning - Items::parseItemNode] Unknown corpseType: " << valueAttribute.as_string() << std::endl;
+					}
+					break;
+				}
+
+				case ITEM_PARSE_CONTAINERSIZE: {
+					it.maxItems = pugi::cast<uint16_t>(valueAttribute.value());
+					break;
+				}
+
+				case ITEM_PARSE_FLUIDSOURCE: {
+					tmpStrValue = asLowerCaseString(valueAttribute.as_string());
+					if (tmpStrValue == "water") {
```

Comment:
why not make this a map?

Reply:
Legacy code. Truly useful maps are a feature of C++11 and the codebase is probably older :P